File: Baekjoon/CPP/10815.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int n, m;


vector<int> card;
vector<int> compare;
vector<int> result;

bool find_value(int value) {

	int left = 0;
	int right = card.size() - 1;

	while (left <= right) {
		int mid = (left + right) / 2;

		if (card[mid] > value) {
			right = mid - 1;
		}
		else if (card[mid] < value) {
			left = mid + 1;
		}
		else {

			return true;
		}
	}

	return false;

}

void solution() {

	cin >> n;

	for (int i = 0; i < n; i++) {

		int temp;

		cin >> temp;

		card.push_back(temp);

	}

	cin >> m;
	

	for (int i = 0; i < m; i++) {
		int temp;
		cin >> temp;
		
		compare.push_back(temp);
	}

	sort(card.begin(), card.end());


	for (int i = 0; i < m; i++) {

		int value = compare[i];

		if (find_value(value)) result.push_back(1);
		else result.push_back(0);
	}
	

	for (int i = 0; i < m; i++) {
		cout << result[i] << ' ';
	}


}

int main() {

	solution();

	return 0;
}

File: Baekjoon/Swift/10815.swift
//
//  main.swift
//  swift_test
//
//  Created by HYUNSEUNG LEE on 2020/10/09.
//

import Foundation

let N = Int(readLine()!)!

var ArrayN = readLine()!.split(separator: " ").map { Int(String($0))! }

let M = Int(readLine()!)!

//var ArrayM = Array(repeating: 0, count: M)
var ArrayM = readLine()!.split(separator: " ").map{ Int(String($0))! }
//let ArrayM = readLine()!.split(separator: " ").map { Int($0)! }

var result = Array<Int>()

func findValue(_ value: Int) -> Bool{
    
    var left = 0
    var right = ArrayN.count - 1
    
    while left <= right{
        
        let mid = (left + right) / 2
        
        if ArrayN[mid] > value{
            right = mid - 1
        }
        else if ArrayN[mid] < value{
           left = mid + 1
        }
        else{
            return true
        }
    }
    return false
    
    
}

ArrayN.sort()

for value in ArrayM{
    if findValue(value){
        result.append(1)
    }
    else{
        result.append(0)
    }
}

for value in result{
    
    print(value, terminator: " ")
}


